src/lang.ts

```diff
@@ -73,14 +73,16 @@ function _mixin(kwArgs: MixinArgs): {} {
  * @param sources Any number of objects whose enumerable own properties will be copied to the target object
  * @return The modified target object
  */
-export function assign(target: {}, ...sources: {}[]): {} {
-	return _mixin({
-		deep: false,
-		inherited: false,
-		sources: sources,
-		target: target
-	});
-}
+export let assign = has('object-assign') ?
+	(<any> Object).assign :
+	function assign(target: {}, ...sources: {}[]): {} {
+		return (typeof (<any> Object).assign === 'function') ? (<any> Object).assign.apply(null, arguments) : _mixin({
+			deep: false,
```

Comment:
Following that logic, wouldn't your `ObjectAssignConstructor` interface then be needed both in the `has` test and here?

I'm usually pro-avoiding-any, but it seems like a whole lot of pointless runaround for something that ultimately doesn't matter in this case (at least as far as type information for the external API is concerned).